packages/astro/src/core/render-context.ts

```diff
@@ -276,6 +276,7 @@ export class RenderContext {
 
 	async createResult(mod: ComponentInstance) {
 		const { cookies, pathname, pipeline, routeData, status } = this;
+		const reactServerActionResult = await getReactServerActionResult(this);
```

Comment:
Is there a reason why this needs to happen here? I think you can probably move this into the integration. 

if the issue is that this needs to be called only once, you can use the `this` value inside of the integration rendering to act like a key. That way you can know it was only called once per render.

We do this inside the Preact integration, see here: https://github.com/withastro/astro/blob/0dbd8eeb77065f3ed03f481c8042f2896a5448c4/packages/integrations/preact/src/server.ts#L61